packages/flutter_tools/lib/src/commands/test.dart

```diff
@@ -31,6 +32,9 @@ class TestCommand extends FlutterCommand {
   String get description => 'Run Flutter unit tests for the current project (Linux only).';
 
   @override
+  bool get shouldRunPub => argResults['flutter-repo'] ? false : super.shouldRunPub;
```

Comment:
you could also write this as `!argResults['flutter-repo'] && super.shouldRunPub`.